src/main/java/htsjdk/tribble/readers/AsciiLineReader.java

```diff
@@ -36,18 +35,36 @@
  * @author jrobinso
  */
 public class AsciiLineReader implements LineReader, LocationAware {
+    private static final Log log = Log.getInstance(AsciiLineReader.class);
+
     private static final int BUFFER_OVERFLOW_INCREASE_FACTOR = 2;
     private static final byte LINEFEED = (byte) ('\n' & 0xff);
     private static final byte CARRIAGE_RETURN = (byte) ('\r' & 0xff);
 
-    private final PositionalBufferedStream is;
+    private PositionalBufferedStream is;
     private char[] lineBuffer;
     private int lineTerminatorLength = -1;
 
+    protected AsciiLineReader() {};
```

Comment:
This is not a great use of an empty constructor. If subclassed improperly a user ends up with a class that's in an invalid state. It will only work correctly if certain methods are overridden, which are not part of a documented API.

The proper solution is to create an abstract base class (or interface with default methods) that both `AsciiLineReader` and `BlockCompressedAsciiLineReader` inherit from. That makes it explicit which methods must be overridden and prevents having inherited fields in `BlockCompressedAsciiLineReader` that are invalid.